code/modules/library/library_admin.dm

```diff
@@ -0,0 +1,212 @@
+#define LIBRARY_MENU_MAIN     1
+#define LIBRARY_MENU_CKEY     2
+#define LIBRARY_MENU_REPORTS  3
+
+/client/proc/library_manager()
+	set name = "Manage Library"
+	set category = "Admin"
+	set desc = "Manage Flagged Books and Perform Maintenance on the Library System"
+
+	if(!check_rights(R_ADMIN))
+		return
+
+	if(!SSticker)
+		to_chat(usr, "<span class='warning'>The game hasn't started yet!</span>")
+		return
+
+	var/datum/ui_module/library_manager/L = new()
+	L.ui_interact(usr)
+
+/datum/ui_module/library_manager
+	name = "Library Manager"
+	///Where we will store our cachedbook datums
+	var/list/cached_books = list()
+	///list of assoc lists detailing each invidual reports, can contain multiple reports for same book
+	var/list/reports = list()
+
+	///TGUI page we are currently on
+	var/page_state = LIBRARY_MENU_MAIN
+	///Ckey's books we are viewing
+	var/selected_ckey
+
+	///information for the book we are opening in browserui
+	var/datum/cachedbook/view_book
+	///browserui helper variable for turning pages in book
+	var/view_book_page = 0
+
+/datum/ui_module/library_manager/ui_interact(mob/user, ui_key = "main", datum/tgui/ui = null, force_open = FALSE, datum/tgui/master_ui = null, datum/ui_state/state = GLOB.admin_state)
+	ui = SStgui.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "LibraryManager", name, 600, 600, master_ui, state)
+		ui.autoupdate = TRUE
+		ui.open()
+
+/datum/ui_module/library_manager/ui_data(mob/user)
+	var/list/data = list()
+	data["pagestate"] = page_state
+	data["booklist"] = cached_books
+	data["ckey"] = selected_ckey ? selected_ckey : "ERROR"
+	data["reports"] = reports
+	for(var/CB in cached_books)
+
+	data["modal"] = ui_modal_data(src)
+	return data
+
+/datum/ui_module/library_manager/ui_act(action, params)
+	if(..())
+		return
+
+	if(ui_act_modal(action, params))
+		return
+
+	switch(action)
+		if("view_reported_books")
+			reports= list()
+			for(var/datum/cachedbook/CB in  GLOB.library_catalog.getFlaggedBooks())
+				for(var/r in CB.reports)
+					var/datum/flagged_book/report = r
+					if(!report)
+						return
+					var/datum/library_category/report_category =  GLOB.library_catalog.getReportCategoryByID(report.category_id)
+					var/report_info = list(
+						"reporter_ckey" = report.reporter,
+						"uploader_ckey" = CB.ckey,
+						"id" = CB.id,
+						"title" = CB.title,
+						"author" = CB.author,
+						"report_description" = report_category.description,
+					)
+					reports += list(report_info)
+			page_state = LIBRARY_MENU_REPORTS
+		if("delete_book")
+			if(GLOB.library_catalog.removeBookByID(text2num(params["bookid"])))
```

Comment:
Might want to check if the param is actually a number before doing this. AA might have a better answer for this though

Reply:
better sanitization added